experimenter/experimenter/experiments/tests/test_models.py

```diff
@@ -2180,10 +2180,43 @@ def _clone_experiment_and_assert_common_expectations(
         return child
 
     def test_get_changelogs(self):
-        experiment = NimbusExperimentFactory.create(slug="experiment-1")
-        generate_nimbus_changelog(experiment, experiment.owner, "Test message")
+        experiment = NimbusExperimentFactory.create(
+            slug="experiment-1",
+            published_dto={"id": "experiment", "test": False},
+        )
+        user = UserFactory.create()
+
+        generate_nimbus_changelog(experiment, user, "created")
+
+        experiment.status = NimbusExperiment.Status.DRAFT
+        experiment.status_next = NimbusExperiment.Status.LIVE
+        experiment.publish_status = NimbusExperiment.PublishStatus.REVIEW
+        experiment.published_dto = {"id": "experiment", "test": True}
+        experiment.save()
+
+        generate_nimbus_changelog(experiment, user, "test message")
```

Comment:
Yeah, I was a bit confused on how to generate the changelogs and make them so that they are reported on different date but couldn't really figure out how I could achieve that. I'll look into it more and try and make the test more robust